components/DownloadButton.tsx

```diff
@@ -0,0 +1,76 @@
+"use client"
+
+import type { Proof, Team } from "@/lib/types"
+
+import ArrowDown from "@/components/svgs/arrow-down.svg"
+
+import { cn } from "@/lib/utils"
+
+import { Button } from "./ui/button"
+import StatusIcon from "./StatusIcon"
+import Tooltip from "./Tooltip"
+
+import { downloadProof } from "@/lib/proofs"
+
+type DownloadButtonProps = {
+  proof: Proof & { team?: Team }
+  className?: string
+}
+
+const DownloadButton = ({ className, proof }: DownloadButtonProps) => {
+  const { proof_status, proof: binary, team } = proof
+  const teamName = team?.team_name ? team.team_name : "Team"
+
+  const sizingClassName =
+    "h-8 gap-2 self-center text-2xl sm:max-md:w-40 lg:w-40"
+  const labelClassName =
+    "hidden text-nowrap text-xs font-bold sm:block md:hidden lg:block"
+  const fakeButtonClassName =
+    "bg-body-secondary/10 hover:bg-body-secondary/10 cursor-auto"
+
+  if (proof_status === "proved")
+    return (
+      <Button
+        variant="outline"
+        className={cn(sizingClassName, className)}
+        size="icon"
+        disabled={!binary}
+        onClick={() => downloadProof(proof)}
+      >
+        <ArrowDown />
+        <span className={labelClassName}>Download proof</span>
+      </Button>
+    )
+
+  if (proof_status === "proving")
+    return (
+      <Tooltip
+        content={`${teamName} currently generating proof for this block`}
+      >
+        <Button size="icon" variant="outline" asChild>
```

Comment:
@pettinarip See here for the fix to your comment from a previous PR... This prevents the repetition of using the ui/button styles as raw string here by just using asChild

Reply:
Great! that works well.